boefjes/boefjes/storage/memory.py

```diff
@@ -50,57 +51,80 @@ def create_boefje(self, boefje: Boefje) -> None:
     def create_normalizer(self, normalizer: Normalizer) -> None:
         self._normalizers[normalizer.id] = normalizer
 
+    def update_boefje(self, boefje_id: str, data: dict) -> None:
+        if not data:
+            return
+
+        if boefje_id not in self._boefjes:
+            raise PluginNotFound(boefje_id)
+
+        boefje = self._boefjes[boefje_id]
+
+        for key, value in data.items():
+            setattr(boefje, key, value)
+
+    def update_normalizer(self, normalizer_id: str, data: dict) -> None:
+        if not data:
+            return
+
+        if normalizer_id not in self._normalizers:
+            raise PluginNotFound(normalizer_id)
+
+        normalizer = self._normalizers[normalizer_id]
+
+        for key, value in data.items():
+            setattr(normalizer, key, value)
+
     def delete_boefje_by_id(self, boefje_id: str) -> None:
         del self._boefjes[boefje_id]
 
     def delete_normalizer_by_id(self, normalizer_id: str) -> None:
         del self._normalizers[normalizer_id]
 
 
-class SettingsStorageMemory(SettingsStorage):
+class ConfigStorageMemory(ConfigStorage):
     def __init__(self):
         self._data = {}
+        self._enabled = {}
 
-    def get_all(self, organisation_id: str, plugin_id: str) -> dict[str, str]:
+    def get_all_settings(self, organisation_id: str, plugin_id: str) -> dict[str, str]:
         if organisation_id not in self._data:
             return {}
 
         return self._data[organisation_id].get(plugin_id, {})
 
-    def upsert(self, values: dict, organisation_id: str, plugin_id: str) -> None:
+    def upsert(
+        self, organisation_id: str, plugin_id: str, settings: dict | None = None, enabled: bool | None = None
+    ) -> None:
         if organisation_id not in self._data:
             self._data[organisation_id] = {}
 
+        if organisation_id not in self._enabled:
+            self._enabled[organisation_id] = {}
+
         if plugin_id not in self._data[organisation_id]:
             self._data[organisation_id][plugin_id] = {}
```

Comment:
For this pattern you can also use the `dict.setdefault` method. Saves a few lines/ if-clauses